examples/zigzag/main.go

```diff
@@ -0,0 +1,272 @@
+// Copyright 2017 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build example jsgo
+
+package main
+
+import (
+	"fmt"
+	"image/color"
+	_ "image/jpeg"
+	"log"
+	"math/rand"
+
+	"github.com/hajimehoshi/ebiten"
+	"github.com/hajimehoshi/ebiten/ebitenutil"
+	"github.com/hajimehoshi/ebiten/inpututil"
+)
+
+const (
+	screenWidth  = 640
+	screenHeight = 480
+)
+
+var (
+	emptyImage *ebiten.Image
+)
+
+// Scale ...
+type Scale struct {
+	X int64
+	Y int64
+}
+
+// Game ...
+type Game struct {
+	moveDirection int8
+	AppleX        int64
+	AppleY        int64
+	scalesLoc     map[int64]*Scale
+	timer         uint64
+	speed         int8
+	Score         int64
+	bestScore     int64
+	Level         int8
+	nearapple     bool
+}
+
+func init() {
+	emptyImage, _ = ebiten.NewImage(1, 1, ebiten.FilterDefault)
+	_ = emptyImage.Fill(color.White)
+}
+
+func (g *Game) collision() bool {
+	if g.scalesLoc[0].X < g.AppleX+10 &&
+		g.scalesLoc[0].X+10 > g.AppleX &&
+		g.scalesLoc[0].Y < g.AppleY+10 &&
+		g.scalesLoc[0].Y+10 > g.AppleY {
+		// collision detected!
+		return true
+	}
+	return false
+}
+
+func (g *Game) nearApple() bool {
+	if g.scalesLoc[0].X < (g.AppleX-30)+60 &&
+		g.scalesLoc[0].X+10 > g.AppleX-30 &&
+		g.scalesLoc[0].Y < (g.AppleY-30)+50 &&
+		g.scalesLoc[0].Y+10 > g.AppleY-30 {
+		// collision detected!
+		return true
+	}
+	return false
+}
+
+func (g *Game) selfCollision() bool {
```

Comment:
collidesWithSelf?

Reply:
collidesWithSelf sounds better name. Thank you.